modules/hostap/CMakeLists.txt

```diff
@@ -208,18 +207,25 @@ zephyr_library_sources_ifdef(CONFIG_WIFI_NM_WPA_SUPPLICANT_AP
 	${HOSTAP_SRC_BASE}/utils/crc32.c
 )
 
+zephyr_library_sources_ifdef(CONFIG_IEEE80211AX
+	${HOSTAP_SRC_BASE}/ap/ieee802_11_he.c
+)
+
 zephyr_library_compile_definitions_ifdef(CONFIG_WIFI_NM_WPA_SUPPLICANT_AP
 	CONFIG_AP
 	CONFIG_NO_RADIUS
 	CONFIG_NO_VLAN
 	CONFIG_NO_ACCOUNTING
 	NEED_AP_MLME
 	CONFIG_IEEE80211AC
-	CONFIG_IEEE80211AX
 	CONFIG_EAP_SERVER
 	CONFIG_EAP_SERVER_IDENTITY
 )
 
+zephyr_library_compile_definitions_ifdef(CONFIG_WIFI_NM_WPA_SUPPLICANT_AP AND CONFIG_WIFI_NM_WPA_SUPPLICAT_11AX
```

Comment:
If we have the `depends on` set, then this could be reduced to
```
zephyr_library_compile_definitions_ifdef(CONFIG_WIFI_NM_WPA_SUPPLICANT_11AX
   CONFIG_IEEE80211AX
)
```

Reply:
Yes